framework/Models/ExternalModel.py

```diff
@@ -156,30 +169,38 @@ def localInputAndChecks(self,xmlNode):
       self.sim = ExternalModel.plugins.returnPlugin("ExternalModel",paramInput.parameterValues['subType'],self)
     else:
       self.raiseAnError(IOError,'"ModuleToLoad" attribute or "subType" not provided for Model "ExternalModel" named "'+self.name+'"!')
+    if not self.pickled:
+      # check if there are variables and, in case, load them
+      for child in paramInput.subparts:
+        if child.getName() =='variable':
+          self.raiseAnError(IOError,'"variable" node included but has been depreciated!  Please list variables in a "variables" node instead.  Remove this message by Dec 2016.')
+        elif child.getName() == 'variables':
+          if len(child.parameterValues) > 0:
+            self.raiseAnError(IOError,'the block '+child.getName()+' named '+child.value+' should not have attributes!!!!!')
+          for var in child.value.split(','):
+            var = var.strip()
+            self.modelVariableType[var] = None
+      # adjust model-aware variables based on aliases
+      self._replaceVariablesNamesWithAliasSystem(self.modelVariableType,'inout')
+      self.listOfRavenAwareVars.extend(self.modelVariableType.keys())
+      # check if there are other information that the external module wants to load
+      #TODO this needs to be converted to work with paramInput
+      if '_readMoreXML' in dir(self.sim):
+        self.sim._readMoreXML(self.initExtSelf,xmlNode)
 
-    # check if there are variables and, in case, load them
-    for child in paramInput.subparts:
-      if child.getName() =='variable':
-        self.raiseAnError(IOError,'"variable" node included but has been depreciated!  Please list variables in a "variables" node instead.  Remove this message by Dec 2016.')
-      elif child.getName() == 'variables':
-        if len(child.parameterValues) > 0:
-          self.raiseAnError(IOError,'the block '+child.getName()+' named '+child.value+' should not have attributes!!!!!')
-        for var in child.value.split(','):
-          var = var.strip()
-          self.modelVariableType[var] = None
-    # adjust model-aware variables based on aliases
-    self._replaceVariablesNamesWithAliasSystem(self.modelVariableType,'inout')
-    self.listOfRavenAwareVars.extend(self.modelVariableType.keys())
-    # check if there are other information that the external module wants to load
-    #TODO this needs to be converted to work with paramInput
-    if '_readMoreXML' in dir(self.sim):
-      self.sim._readMoreXML(self.initExtSelf,xmlNode)
+  def evaluate(self, request: dict):
```

Comment:
I wonder if we could find some naming conventions that aren't purely synonyms to make the distinction between `run` and `evaluate` more clear. However, such renaming should probably be saved for a general Model serialization rollout, in my opinion.